### ISSUE TITLE
Loss around 75 and 0 boxes on inference.

### ISSUE THREAD

**Issue by skier233** [qqwweee/keras-yolo3#531]
My validation loss stopped improving much around 75 so I stopped training and tried inference. However, when doing inference, no boxes are ever reported. I have 20 classes so I would expect at least one box to show up somewhere even if a false positive. I am getting these errors at the beginning of training that seem to be complaining about the model structure even though I converted it from darknet exactly as the readme instructed. I'm not sure if these could be the issue:
`Using TensorFlow backend.
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorflow\python\framework\dtypes.py:516: FutureWarning: Passing (type, 1)
or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorflow\python\framework\dtypes.py:517: FutureWarning: Passing (type, 1)
or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorflow\python\framework\dtypes.py:518: FutureWarning: Passing (type, 1)
or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorflow\python\framework\dtypes.py:519: FutureWarning: Passing (type, 1)
or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorflow\python\framework\dtypes.py:520: FutureWarning: Passing (type, 1)
or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorflow\python\framework\dtypes.py:525: FutureWarning: Passing (type, 1)
or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorboard\compat\tensorflow_stub\dtypes.py:541: FutureWarning: Passing (ty
pe, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorboard\compat\tensorflow_stub\dtypes.py:542: FutureWarning: Passing (ty
pe, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorboard\compat\tensorflow_stub\dtypes.py:543: FutureWarning: Passing (ty
pe, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorboard\compat\tensorflow_stub\dtypes.py:544: FutureWarning: Passing (ty
pe, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorboard\compat\tensorflow_stub\dtypes.py:545: FutureWarning: Passing (ty
pe, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
C:\Users\tyler\Desktop\Coding\Testing\TensorFlowNeuralNet\yoloenv\lib\site-packages\tensorboard\compat\tensorflow_stub\dtypes.py:550: FutureWarning: Passing (ty
pe, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])`

**Comment by remingm** [comment]
We're having similar issues at #512

**Comment by skier233** [comment]
So when I trained the model mentioned above, I used images scaled to 1088,1088. However, I used the original anchors provides by yolo which mightve been the issue. I’ve since tried using 416,416 images and I’m getting good results. However, does anyone know what I need to do to scale the anchors or how I would come up with my own anchors?

**Comment by datduonguva** [comment]
@skier233 run the kmeans.py

**Comment by skier233** [comment]
@datduonguva Thank you I will try that.

**Comment by LoaferCH** [comment]
@skier233 Did kmeans.py help? I have the similar issue. Loss is around 15 and still 0 boxes.

**Comment by skier233** [comment]
I have boxes. I got them before after some tweaking. however, running kmeans.py did not seem to help. Using 1088p images, I have a validation loss of 60 meanwhile using the exact same images at 416p, I got a validation loss of 38 and the annotations were much better. I’m a bit confused why feeding it higher resolution images is leading to lower mAP. When I trained a model using retinanet with this data, my mAP was abysmal on anything under 720p

**Comment by Andiez-Nguyen** [comment]
> @skier233 run the kmeans.py

my most of data is 1920x1080,by the kmeans the anchors is bigger than 416x416 while in data_generator the original image will resize to 416x416 ,is this normal?

**Comment by jwnirvana** [comment]
> I have boxes. I got them before after some tweaking. however, running kmeans.py did not seem to help. Using 1088p images, I have a validation loss of 60 meanwhile using the exact same images at 416p, I got a validation loss of 38 and the annotations were much better. I’m a bit confused why feeding it higher resolution images is leading to lower mAP. When I trained a model using retinanet with this data, my mAP was abysmal on anything under 720p

Hello, could you please tell me how much the average  iou  is after running kmeans.py?

**Comment by jwnirvana** [comment]
> @skier233 Did kmeans.py help? I have the similar issue. Loss is around 15 and still 0 boxes.

Hello, could you please tell me how much the average iou is after running kmeans.py? I run it ,but the iou is so small. I am so confused.
